protocol-specs/generators/vote_transaction/index.js

```diff
@@ -0,0 +1,517 @@
+		address: '9439340122733729158L',
+	},
+	{
+		passphrase:
+			'shield almost dinner rebel rotate nut harvest candy battle fix pass nut',
+		publicKey:
+			'e1f20a8b1c64193db5f009fd4d88fde9bd1320b8c921fafe800bacd94c347a2b',
+		address: '15206119636421553919L',
+	},
+	{
+		passphrase:
+			'lounge basket time economy lounge destroy organ dynamic save auction loud secret',
+		publicKey:
+			'1eb301328a5681a4d3a002c892644efcc057436985d48d55261133dae0af5c41',
+		address: '10189413624252937509L',
+	},
+	{
+		passphrase:
+			'entire jungle toilet remain zoo spread combine eternal rug wish display infant',
+		publicKey:
+			'0355085d4d6cc2565c69a248846e9d1cb7af023f8d3a2b31445a0386a45758a4',
+		address: '15682953477545527099L',
+	},
+	{
+		passphrase:
+			'upset ivory pigeon dash theory lonely arch flock wrap adapt enable runway',
+		publicKey:
+			'f740f22ff4413757457cd25b390f5312b5b10dd09f4ed901848a57cb84bc1261',
+		address: '4458741937615618075L',
+	},
+	{
+		passphrase:
+			'search wild flavor suit culture alcohol energy rate glad trophy angle promote',
+		publicKey:
+			'09bf0bd593f354f7949cbbf42cedfdc9fabd2d7da5ff24e0f24c4017ebdb7450',
+		address: '10182969975768460850L',
+	},
+	{
+		passphrase:
+			'quality sniff spice melody royal wide industry parent antique animal inquiry economy',
+		publicKey:
+			'2998ae5c6b28388fd654262ca19a4d669abf067aa2a28fa2ecb94079d1386ec9',
+		address: '16218061708783968021L',
+	},
+	{
+		passphrase:
+			'find alcohol buzz emotion holiday forest problem age multiply sadness hen fashion',
+		publicKey:
+			'f8b282fe76bed11e0048f668e2768f1b5346acd77b3afe2a01c9b3874612fba2',
+		address: '12502596496028234907L',
+	},
+	{
+		passphrase:
+			'purse erase first gallery drama horror gloom abandon cupboard pill twist bitter',
+		publicKey:
+			'19528c41f749fb0acd840b5349823afea8d96d9380cf4c674a5cf522417a6946',
+		address: '9098130216893659918L',
+	},
+];
+
+const getAssetBytes = asset => {
+	const buffers = [];
+	for (const vote of asset.votes) {
+		const addressBuffer = Buffer.alloc(8);
+		addressBuffer.writeBigUInt64BE(BigInt(vote.delegateAddress.slice(0, -1)));
+		buffers.push(addressBuffer);
+		const amountBuffer = Buffer.alloc(8);
+		amountBuffer.writeBigInt64BE(BigInt(vote.amount));
+		buffers.push(amountBuffer);
+	}
+
+	return Buffer.concat(buffers);
+};
+
+const getSignBytes = tx => {
+	const transactionNonce = intToBuffer(tx.nonce, 8);
+	const buf = Buffer.concat([
+		Buffer.alloc(1, tx.type),
+		transactionNonce,
+		hexToBuffer(tx.senderPublicKey),
+		intToBuffer(tx.fee.toString(), 8),
+		getAssetBytes(tx.asset),
+	]);
+
+	return buf;
+};
+
+const getId = transactionBytes => {
+	const transactionHash = hash(transactionBytes);
+	const bufferFromFirstEntriesReversed = getFirstEightBytesReversed(
+		transactionHash,
+	);
+	const transactionId = bufferToIntAsString(bufferFromFirstEntriesReversed);
+
+	return transactionId;
+};
+
+const generateValidUpvoteTransaction = () => {
+	const unsignedTransaction = {
+		type: 13,
```

Comment:
isn't this is type 11 type of transaction?

Reply:
yes, it should be 11, but in order to make it non-breaking, I set it to 13 for now.
Ww can change it later once all the DPOS change is done